### ISSUE TITLE
set the selected row after updateReactable

### ISSUE THREAD

**Issue by paulvanoppen** [glin/reactable#128]
Is it possible to set the selected row using updateReactable and its selected parameter after updating the data?
I obtain the selected row using getReactableState("table-id", "selected") and assign that to a reactiveValues object.
This value is then passed to the updateReactable function but there are two different responses:

1. If the data has not changed (so no user input), the correct row is selected after update Reactable.
2. If the data is changed, the selected row is not updated.

Code snippet:
```
    row_selected <- reactive(getReactableState("table-id", "selected"))
    row_selected_val <- reactiveValues(value = NULL)
   
    observeEvent(row_selected(), {
      row_selected_val$value <- row_selected()
    })

   observeEvent(input$ok, {
   # a modalDialog allows for new data input. There is an 'ok' actionButton and this code is executed upon click of the actionbutton
      updateReactable(outputId = "table-id", 
                      data = my_object$data,  # this is a reactiveValues object which can be changed by the user of the shiny app
                      expanded = TRUE,
                      selected = row_selected_val$value,
                      page = 1)
  })
```

Any suggestions would be helpful.

**Comment by paulvanoppen** [comment]
A bit more information about this. If I do:

```
      updateReactable(outputId = "table-id", 
                      data =my_object$data, 
                      expanded = TRUE, 
                      session = session)
      
      updateReactable(outputId = "table-id", 
                      expanded = TRUE, 
                      selected = 2,  # force row 2 to be selected
                      session = session)
```
I can observe:

1.  The data gets updated in the table at the client side
2.  After data update no row is selected
3.  For a brief moment, row 2 is selected
4. . Row 2 is unselected 
How do I keep the selected row selected?
I am running R 4.0.3,  reactable 0.2.3.9000, reactR 0.4.4.

**Comment by paulvanoppen** [comment]
it may not be a reactable issue: closer inspection shows that the deselection of the row occurs after the shiny function removeModal has been called.  All removeModal does is:

```
function (session = getDefaultReactiveDomain()) 
{
    session$sendModal("remove", NULL)
}
```
Does session$sendModal("remove", NULL) affect reactable?

**Comment by paulvanoppen** [comment]
I found the cause of this issue and it was unrelated to reactable: as I pass a reactiveValues object to reactable, reactable takes a dependency on this reactive object. In the code for the 'ok' actionbutton I update the reactiveValues object with the values entered in the modal Dialog (user input to change values in the table). This causes output$"table-id" to be re-evaluated. I did not have a 'defaultSelected' parameter defined in the reactable code and as a result I lost the selected row. 
In other words: updating the reactiveValues object causes the reactable table to be redrawn each and every time. The fact that I called updateReactable afterwards did not matter.

I resolved the issue by passing the isolated value of the selected row, stored in its own reactiveValues object, to the 'defaultSelected' parameter of reactable.

I am a little concerned that this may not be the best solution As I now redraw the reactable table after each change. As my table is small (~25 rows) it's ok but there would be performance issues with larger tables.  The purpose of the Shiny application is to edit the data in the table, so the data object has to be reactive (I think).
